src/content/components/GlobalContext/GlobalContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Iterations } from "../../../common/IterationLookup";
+import { QueryManager } from "../../lib/utils";
+
+export const GlobalContext = React.createContext({} as GlobalContextProps);
+
+export interface MetaBug {
+  id: string;
+  component?: string;
+  priority?: string;
+  displayName?: string;
+}
+
+interface GlobalContextProviderProps {
+  metas: MetaBug[];
+  iterations: Iterations;
+  qm: QueryManager;
+}
+
+export interface GlobalContextProps extends GlobalContextProviderProps {
+  setMetas: (metas: MetaBug[]) => void;
+  setIterations: (iterations: Iterations) => void;
+}
+
+export class GlobalContextProvider extends React.Component<
+  Readonly<GlobalContextProviderProps>
+> {
+  declare state: GlobalContextProps;
+
+  constructor(props: Readonly<GlobalContextProviderProps>) {
+    super(props);
+    this.state = {
+      metas: props.metas,
+      iterations: props.iterations,
+      qm: props.qm,
+      setMetas: (metas: MetaBug[]) => {
+        this.setState({ metas });
+      },
+      setIterations: (iterations: Iterations) => {
+        this.setState({ iterations });
+      },
+    };
+  }
+
+  render() {
+    return (
+      <GlobalContext.Provider value={this.state}>
+        {this.props.children}
+      </GlobalContext.Provider>
+    );
+  }
+}
```

Comment:
This can be a functional component.

```ts
export function GlobalContextProvider({metas, iterations, qm, children}: Readonly<GlobalContextProviderProps>) {
  const [metas, setMetas] = useState<MetaBug[]>(metas);
  const [iterations, setIterations] = useState<Iterations>(iterations);

  const value = useMemo(() => ({
    metas,
    iterations,
    setMetas,
    setIterations,
    qm,
  }, [metas, setMetas, iterations, setIteartions, qm]);

  return (<GlobalContext.Provider value={value}>{children}</GlobalContext.Provider>);
}
```

Reply:
Are you sure it doesn't cause everything to re-render every time the state changes? The reason I used a class component was because I read something about that. If you're not sure I can do some testing to find out.

Reply:
State changes will cause re-renders in your class based component. If it didnt re-render, how would children get updated context?

Reply:
Err not the context's own state, but the state of the context's parent. I found the article I was thinking of: https://reactjs.org/docs/context.html#caveats